### PR TITLE
OCC-277: Get stripe confirm parameters async

### LINKED ISSUE #476: Abstract the Stripe GetConfirmPaymentParameters logic to service (OCC-277)
### Is your feature request related to a problem? Please describe

abstract the logic inside service

### Describe the solution you'd like

use new method of GetStripeConfirmParametersAsync

[Jira issue](https://lombiq.atlassian.net/browse/OCC-277)

### PR DESCRIPTION
**Pull request by infofromca**
Fixes #476

### CODE REVIEW

File: src/Modules/OrchardCore.Commerce.Payment.Stripe/Services/StripePaymentService.cs [hunk 1]
```diff
@@ -215,9 +216,51 @@ public async Task<PaymentIntent> CreatePaymentIntentAsync(Amount total)
...
+            Line1 = address.StreetAddress1 ?? string.Empty,
+            Line2 = address.StreetAddress2 ?? string.Empty,
+            PostalCode = address.PostalCode ?? string.Empty,
+            State = address.Province ?? string.Empty,
+        };
     private async Task<PaymentIntent> GetOrUpdatePaymentIntentAsync(
-        string paymentIntentId,
-        Amount defaultTotal)
+    string paymentIntentId,
+    Amount defaultTotal)
```
Comment (sarahelsaig) [inline src/Modules/OrchardCore.Commerce.Payment.Stripe/Services/StripePaymentService.cs:263]: Revert this change, parameters should be indented from the starting line of the function signature.

File: src/Modules/OrchardCore.Commerce.Payment.Stripe/Abstractions/IStripePaymentService.cs [hunk 2]
```diff
@@ -61,4 +61,11 @@ public interface IStripePaymentService
         Controller controller,
         ContentItem order,
         PaymentIntent paymentIntent);
+
+    /// <summary>
+    /// Get the confirmation parameters for Stripe.
+    /// </summary>
+    /// <param name="middlewareUrl">The url for the middleware of Stripe.</param>
+    /// <returns><see cref="PaymentIntentConfirmOptions"/>.</returns>
```
Comment (sarahelsaig) [inline src/Modules/OrchardCore.Commerce.Payment.Stripe/Abstractions/IStripePaymentService.cs:69]:
```suggestion
```
This return text adds no useful information, so it can be removed.

File: src/Modules/OrchardCore.Commerce.Payment.Stripe/Abstractions/IStripePaymentService.cs [hunk 3]
```diff
@@ -61,4 +61,11 @@ public interface IStripePaymentService
...
+    Task<PaymentIntentConfirmOptions> GetStripeConfirmParametersAsync(string middlewareUrl);
```
Comment (sarahelsaig) [inline src/Modules/OrchardCore.Commerce.Payment.Stripe/Abstractions/IStripePaymentService.cs:70]: Shouldn't the parameter type be `Uri` instead?

**Review by sarahelsaig — CHANGES_REQUESTED** [review]
Thanks, looks good. I just need a few small tweaks before it can be merged.

File: src/Modules/OrchardCore.Commerce.Payment.Stripe/Abstractions/IStripePaymentService.cs (hunk 3 above)
Comment (infofromca) [reply src/Modules/OrchardCore.Commerce.Payment.Stripe/Abstractions/IStripePaymentService.cs:69]: not necessary in this case, and not increase complicated

Comment (sarahelsaig) [reply src/Modules/OrchardCore.Commerce.Payment.Stripe/Abstractions/IStripePaymentService.cs:69]: After looking the code over again, this has to be absolute URL. Normally in ASP.Net Core you pass around relative virtual URLs, like `~/checkout/middleware/Stripe`. Which wouldn't work in this case because the full URL is sent out to a remote server.
Please rename the parameter to `middlewareAbsoluteUrl`

**Review by sarahelsaig — CHANGES_REQUESTED** [review]

File: src/Modules/OrchardCore.Commerce.Payment.Stripe/Abstractions/IStripePaymentService.cs (hunk 3 above)
Comment (infofromca) [reply src/Modules/OrchardCore.Commerce.Payment.Stripe/Abstractions/IStripePaymentService.cs]: renamed to middlewareAbsoluteUrl
